### PR TITLE
LPS-106345 | Edit experience priority & LPS-106346 | Edit experience

### PR DESCRIPTION
**Pull request by dgarciasarai**
Review from https://github.com/p2kmgcl/liferay-portal/pull/490/commits/ac071a93dd3991ef86cd024d3e4cd29baaf59456 to https://github.com/p2kmgcl/liferay-portal/pull/490/commits/cd2b0b152a369e7bdd701a6b54112a91b189e1e3 (LPS-106345 and LPS-106346).
The other commits are necessary to try these changes (create experience).
Thanks!

### CODE REVIEW

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/components/ExperienceSelector.js [hunk 1]
```diff
@@ -37,6 +36,42 @@ import {
...
+/**
+ * It produces an object with a target and subtarget keys indicating what experiences
+ * should change to change the priority of target priority.
+ */
+function getUpdateExperiencePriorityTargets(
+	orderedExperiences,
+	targetExperienceId,
+	direction
+) {
+	const targetIndex = orderedExperiences.findIndex(
```
[the remark this answers is not in the document]
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/components/ExperienceSelector.js]: Síiiiiiiii 🎉🥩
Screenshot 2020-01-03 at 14 06 20

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js [hunk 2]
```diff
@@ -172,4 +185,234 @@ describe('ExperienceToolbarSection', () => {
 		getByText('experience-locked');
 		getByText('edit-is-not-allowed-for-this-experience');
 	});
+
+	it('calls the backend to increase priority', async () => {
+		window.Liferay.Service = jest.fn(() => Promise.resolve());
```
[the remark this answers is not in the document]
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js]: We'll have to double check

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/API.js [hunk 3]
```diff
@@ -16,12 +16,17 @@ import React from 'react';
 
 export const APIContext = React.createContext({});
 
+// TODO grab urls from displayContext
```
[the remark this answers is not in the document]
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/API.js]: I just created a subtask to move this as a refactor, so our rebase train keeps going.  🚂

**Comment by andresfulla** [comment]
Reviewing! <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/index.js [hunk 4]
```diff
@@ -139,8 +171,8 @@ export default class Experience {
 						action.payload
 					);
 					break;
-				case DELETE_SEGMENTS_EXPERIENCE:
-					nextState = deleteExperienceReducer(
+				case UPDATE_SEGMENTS_EXPERIENCE_PRIORITY:
+					nextState = updateExperiencePriorityReducer(
```
[the remark this answers is not in the document]
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/plugins/experience/index.js]: We'll move this at the end of the task too.

File: modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (andresfulla) [reply modules/apps/layout/layout-content-page-editor-web/test/page_editor/app/plugins/experience/components/ExperienceToolbarSection.test.js]: I won't get to any other file. So I'll leave as it is.

**Pull request closed without merging**